broker-core/src/main/java/io/zeebe/broker/subscription/message/state/Message.java

```diff
@@ -0,0 +1,195 @@
+
+import io.zeebe.util.buffer.BufferReader;
+import io.zeebe.util.buffer.BufferWriter;
+import io.zeebe.util.sched.clock.ActorClock;
+import java.nio.ByteOrder;
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+
+public final class Message implements BufferWriter, BufferReader {
+
+  private final DirectBuffer name = new UnsafeBuffer();
+  private final DirectBuffer correlationKey = new UnsafeBuffer();
+  private final DirectBuffer payload = new UnsafeBuffer();
+  private final DirectBuffer id = new UnsafeBuffer();
+  private long timeToLive;
+  private long deadline;
+  private long key;
+
+  public Message() {}
+
+  public Message(
+      DirectBuffer name,
+      DirectBuffer correlationKey,
+      DirectBuffer payload,
+      DirectBuffer id,
+      long timeToLive) {
+    this.name.wrap(name);
+    this.correlationKey.wrap(correlationKey);
+    this.payload.wrap(payload);
+    this.id.wrap(id);
+
+    this.timeToLive = timeToLive;
+    this.deadline = ActorClock.currentTimeMillis() + timeToLive;
+  }
+
+  Message(
+      final String name, final String correlationKey, final String payload, final long timeToLive) {
+    this(
+        new UnsafeBuffer(name.getBytes()),
+        new UnsafeBuffer(correlationKey.getBytes()),
+        new UnsafeBuffer(payload.getBytes()),
+        new UnsafeBuffer(new byte[0]),
+        timeToLive);
+  }
+
+  Message(
+      final String id,
+      final String name,
+      final String correlationKey,
+      final String payload,
+      final long timeToLive) {
+    this(
+        new UnsafeBuffer(name.getBytes()),
+        new UnsafeBuffer(correlationKey.getBytes()),
+        new UnsafeBuffer(payload.getBytes()),
+        new UnsafeBuffer(id.getBytes()),
+        timeToLive);
+  }
+
+  public DirectBuffer getName() {
+    return name;
+  }
+
+  public DirectBuffer getCorrelationKey() {
+    return correlationKey;
+  }
+
+  public DirectBuffer getPayload() {
+    return payload;
+  }
+
+  public DirectBuffer getId() {
+    return id;
+  }
+
+  public long getTimeToLive() {
+    return timeToLive;
+  }
+
+  public long getDeadline() {
+    return deadline;
+  }
+
+  public long getKey() {
+    return key;
+  }
+
+  public void setKey(long key) {
+    this.key = key;
+  }
+
+  @Override
+  public void wrap(final DirectBuffer buffer, int offset, final int length) {
+    offset = readIntoBuffer(buffer, offset, name);
+    offset = readIntoBuffer(buffer, offset, correlationKey);
+    offset = readIntoBuffer(buffer, offset, payload);
+    offset = readIntoBuffer(buffer, offset, id);
+
+    timeToLive = buffer.getLong(offset, ByteOrder.LITTLE_ENDIAN);
+    offset += Long.BYTES;
+    deadline = buffer.getLong(offset, ByteOrder.LITTLE_ENDIAN);
+    offset += Long.BYTES;
+    key = buffer.getLong(offset, ByteOrder.LITTLE_ENDIAN);
+  }
+
+  static int readIntoBuffer(final DirectBuffer buffer, int offset, final DirectBuffer valueBuffer) {
```

Comment:
Maybe this should go in BufferUtil instead? Since it's not specific to `Message`. Ok maybe BufferUtil isn't also the right place, since this is more specific to direct buffer serialization. But I can imagine we'll be reusing this in other places for RocksDB...

If you can think of a good name for this we can already abstract it somewhere else, but otherwise I'm ok with waiting until we make more use of it to extract it.

Reply:
Yeah I think you are right that we will need this in other places as well. Have to think where to put it

Reply:
Lets postpone this for now until we really need this elsewhere